### DISCUSSION TITLE
Which IR should users use?

### DISCUSSION THREAD

**Discussion by randyzhao** [mlc-ai/mlc-en#14] [Q&A]
TVM seems to introduce several IRs: Relay, Tensor Expression and Tensor IR. If I need to implement a custom op in my model, which IR should I use?

**Comment by yzh119** [comment]
Relay is computational graph level IR so you don't need it if you just want to write an operator.
Tensor Expression would be lowered to Tensor IR, so either would work (Tensor IR can express more operations actually).
